scripts/zkai_d128_range_policy_discipline_gate.py

```
@@ -0,0 +1,777 @@
+    except (UnicodeDecodeError, json.JSONDecodeError) as err:
+        raise D128RangePolicyError(f"failed to parse source evidence {path}: {err}") from err
+    if not isinstance(payload, dict):
+        raise D128RangePolicyError(f"source evidence must be an object: {path}")
+    return payload
+
+
+def validate_source_payload(source: dict[str, Any], spec: SourceSpec) -> None:
+    expect_equal(source.get("schema"), spec.schema, f"{spec.source_id} schema")
+    expect_equal(source.get("decision"), spec.decision, f"{spec.source_id} decision")
+    if str(source.get("target_id", "")).startswith("rmsnorm-swiglu-residual") is False:
+        raise D128RangePolicyError(f"{spec.source_id} target_id drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)